manifests/extract.pp

```diff
@@ -41,25 +41,30 @@
   $strip_components=0,
   $purge=false,
   $user=undef,
-  $tar_command=$::osfamily ? {
-    'Solaris' => 'gtar',
-    default   => 'tar',
-  },
+  $tar_command=undef,
 ) {
 
+  require ::archive::params
+
   if $root_dir {
     $extract_dir = "${target}/${root_dir}"
   } else {
     $extract_dir = "${target}/${name}"
   }
 
+  if $tar_command {
+    $real_tar_command = $tar_command
+  } else {
+    $real_tar_command = $::archive::params::tarcmd
+  }
+
```

Comment:
IIRC, defined types are not able to inherit, only classes can inherit.

Because defined types can not inherit, they can not make use of values from other classes as default values for defined type parameters.